File: backend/app/schemas/notification.py
from __future__ import annotations
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    description: Optional[str]
    type: str
    is_read: bool
    related_activity_id: Optional[uuid.UUID]
    created_at: datetime

File: frontend/src/@types/notification.ts
export interface AppNotification {
  id: string;
  user_id: string;
  title: string;
  description: string | null;
  type: string; // "task" | "log" | "message"
  is_read: boolean;
  related_activity_id: string | null;
  created_at: string;
}
